pkg/test/framework/components/ingress/ingress.go

```diff
@@ -30,12 +39,21 @@ type Instance interface {
 	Address() string
 
 	//  Call makes an HTTP call through ingress, where the URL has the given path.
-	Call(path string) (CallResponse, error)
-	CallOrFail(t test.Failer, path string) CallResponse
+	Call(path, host string) (CallResponse, error)
+	CallOrFail(t test.Failer, path, host string) CallResponse
 }
 
 type Config struct {
 	Istio istio.Instance
+	// IngressType specifies the type of ingress gateway.
+	IngressType GatewayType
```

Comment:
This doesn't look like the right place to put these parameters. It looks like Call method is the right place to specify theser.

Reply:
The IngressType is used by newKube in ingress/kube.go, to determine what address should be returned to c.address. Moving IngressType into Call method will delay the initialization of c.address.